src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -3425,6 +3425,35 @@ class SymbolTableVisitor : public CommonVisitor<SymbolTableVisitor> {
                         throw SemanticError("Decorator: " + name + " is not supported",
                             x.base.base.loc);
                     }
+                } else if (AST::is_a<AST::Call_t>(*dec)) {
+                    AST::Call_t *call_d = AST::down_cast<AST::Call_t>(dec);
+                    if (AST::is_a<AST::Name_t>(*call_d->m_func)) {
+                        std::string name = AST::down_cast<AST::Name_t>(call_d->m_func)->m_id;
+                        if (name == "ccall") {
+                            current_procedure_abi_type = ASR::abiType::BindC;
+                            current_procedure_interface = true;
+                            if (call_d->n_keywords > 0) {
+                                for (size_t i=0; i < call_d->n_keywords; i++) {
+                                    if (std::string(call_d->m_keywords[i].m_arg) == "header") {
+                                        if (AST::is_a<AST::ConstantStr_t>(*call_d->m_keywords[i].m_value)) {
+                                            std::string header_name = AST::down_cast<AST::ConstantStr_t>(
+                                                        call_d->m_keywords[i].m_value)->m_value;
+                                            bindc_name = s2c(al, header_name);
```

Comment:
I'm using `bindc_name` to test the prototype and use it as a header file name which in LLVM is a function name to be called from C. Once this works, do we want to add a new argument in the function type to check which header file it is contained in?

Reply:
I think we do. Probably not in FunctionType but in Function.

Reply:
I don't think Function has a mechanism to store the c header file. It just has bindc_name for its interop with C.
```
| Function(symbol_table symtab, identifier name, ttype function_signature,
        identifier* dependencies, expr* args, stmt* body, expr? return_var,
        access access, bool deterministic, bool side_effect_free)
```

Should we go with this?
```diff
diff --git a/src/libasr/ASR.asdl b/src/libasr/ASR.asdl
index 34af5be3c..3f4b29b4a 100644
--- a/src/libasr/ASR.asdl
+++ b/src/libasr/ASR.asdl
@@ -363,7 +363,7 @@ ttype
     | CPtr()
     | TypeParameter(identifier param, dimension* dims)
     | FunctionType(ttype* arg_types, ttype? return_var_type,
-        abi abi, deftype deftype, string? bindc_name, bool elemental,
+        abi abi, deftype deftype, string? bindc_name, string? c_header, bool elemental,
         bool pure, bool module, bool inline, bool static, ttype* type_params,
         symbol* restrictions, bool is_restriction)
```